components/core/src/GlobalMySQLMetadataDB.cpp

```diff
@@ -197,10 +197,16 @@ void GlobalMySQLMetadataDB::update_metadata_for_files (const std::string& archiv
     }
 }
 
-GlobalMetadataDB::ArchiveIterator* GlobalMySQLMetadataDB::get_archive_iterator () {
-    auto statement_string = fmt::format("SELECT {} FROM {}{} ORDER BY {} ASC, {} ASC", streaming_archive::cMetadataDB::Archive::Id, m_table_prefix,
-                                        streaming_archive::cMetadataDB::ArchivesTableName, streaming_archive::cMetadataDB::Archive::CreatorId,
-                                        streaming_archive::cMetadataDB::Archive::CreationIx);
+GlobalMetadataDB::ArchiveIterator* GlobalMySQLMetadataDB::get_archive_iterator (epochtime_t begin_ts, epochtime_t end_ts) {
+    auto statement_string = fmt::format("SELECT DISTINCT {}{}.{} FROM {}{} JOIN {}{} ON {}{}.{} = {}{}.{} WHERE {}{}.{} > {} AND {}{}.{} < {} ORDER BY {} ASC, {} ASC",
```

Comment:
```suggestion
    auto statement_string = fmt::format("SELECT DISTINCT {}{}.{} FROM {}{} JOIN {}{} ON {}{}.{} = {}{}.{} WHERE {}{}.{} >= {} AND {}{}.{} <= {} ORDER BY {} ASC, {} ASC",
```

Reply:
will need more changes than just `>=` and `<=`. Updated it in my commit